flink-table/flink-table-common/src/main/java/org/apache/flink/table/expressions/SqlExpression.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.expressions;
+
+import org.apache.flink.annotation.Internal;
+import org.apache.flink.table.types.DataType;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+
+import static org.apache.flink.util.Preconditions.checkNotNull;
+
+/**
+ * Expression that wraps a SQL expression string, e.g. {@code date_format(`f0`, 'yyyy-MM-dd')}.
+ * The SQL expression string can be used as serialized string and can be used for storing in Catalog.
+ *
+ * <p>However, the SQL expression string will not be verified when constructing a {@link SqlExpression}
+ * instance. The output type of the SQL expression, i.e. {@link #getOutputDataType()}, is also not be validated.
+ * Users should make sure they are correct.
+ *
+ * <p>NOTE: This is only for internal usage. It will be removed once we fully support conversion
+ * from {@link ResolvedExpression} to SQL string.
+ */
+@Internal
+public class SqlExpression implements ResolvedExpression {
```

Comment:
unnecessary class in this PR?

Reply:
Good catch! I will introduce it in the following PR.